Src/IronPython/Modules/Builtin.cs

```diff
@@ -1013,51 +1038,74 @@ public class mapobject : IEnumerable {
                 int retIndex = 0;
                 if (args.Length == 1) {
                     return min(context, args[retIndex], dict);
                 }
-                object method = GetMinKwArg(dict);
+                
                 object retValue = PythonCalls.Call(context, method, args[retIndex]);
                 PythonContext pc = context.LanguageContext;
 
                 for (int i = 1; i < args.Length; i++) {
                     object tmpRetValue = PythonCalls.Call(context, method, args[i]);
-
+                    
                     if (pc.LessThan(tmpRetValue, retValue)) {
                         retIndex = i;
                         retValue = tmpRetValue;
                     }
                 }
                 return args[retIndex];
             } else {
-                throw PythonOps.TypeError("min expecting 1 arguments, got 0");
+                throw PythonOps.TypeError("min expected 1 arguments, got 0");
             }
         }
 
-        private static object GetMinKwArg([ParamDictionary]IDictionary<object, object> dict) {
-            if (dict.Count != 1)
-                throw PythonOps.TypeError(" min() should have only 1 keyword argument, but got {0} keyword arguments", dict.Count);
+        private static Tuple<object,object> GetMinKwArg([ParamDictionary]IDictionary<object, object> dict, bool isDefaultAllowed) {
+            if (dict.Count != 1 && dict.Count != 2)
+                throw PythonOps.TypeError("min() should have only 2 keyword arguments, but got {0} keyword arguments", dict.Count);
 
+            if (dict.Keys.Contains("default") && !isDefaultAllowed)
+                throw PythonOps.TypeError("Cannot specify a default for min() with multiple positional arguments");
+            
             return VerifyKeys("min", dict);
         }
 
-        private static object VerifyKeys(string name, IDictionary<object, object> dict) {
-            object value;
-            if (!dict.TryGetValue("key", out value)) {
-                ICollection<object> keys = dict.Keys;
-                IEnumerator<object> en = keys.GetEnumerator();
-                if (en.MoveNext()) {
-                    throw PythonOps.TypeError(" {1}() got an unexpected keyword argument ({0})", en.Current, name);
-                }
+        private static Tuple<object, object> VerifyKeys(string name, IDictionary<object, object> dict) {
+            object method;
+            object def;
+
+            int cnt = 0;
+            if (dict.TryGetValue("key", out method)) {
+                cnt++;
             }
-            return value;
+
+            if (dict.TryGetValue("default", out def)) {
+                cnt++;
+            }
+
+            if (dict.Count > 2) {
+                throw PythonOps.TypeError("{1}() takes at most 2 keyword arguments ({0} given)", dict.Count, name);
+            }
+
+            var keys = dict.Keys.Where(x => (string)x != "default" && (string)x != "key");
```

Comment:
Almost there, maybe something like:
```C#

if (dict.Count > cnt) {
    var key = dict.Keys.Cast<string>().Where(x => x != "default" && x != "key").First();
    throw PythonOps.TypeError("{1}() got an unexpected keyword argument ({0})", key, name);
}
```

Reply:
You can remove this line since keys is no longer used.

Reply:
The whole block?

Reply:
Just the line with `var keys = ...`. Unless it does something and I'm not seeing it?

Reply:
My only concern here is being able to throw the error about the first extra key. if it is unnecessary I can take it out?

Reply:
`var key = ...` inside the `if` does that. The `var keys = ...` before the `if` is redundant.

Reply:
ah ok I confused myself. thanks!